client/src/components/projects/project-filter/ProjectFilter.vue

```diff
@@ -0,0 +1,178 @@
+                        v-on:update:checked="handleFilterAction(program, 'program')"/> 
+              <span class="title is-5 is-size-16">{{ program.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+    <Accordeon
+          name="Project type"
+          :opened="typeAccordionOpened"
+          @update:opened="typeAccordionOpened=!typeAccordionOpened"
+        >
+          <div
+            v-for="t in types"
+            :key="t.name"
+          >
+            <div class="filter-item">
+              <Checkbox :checked = "t.checked"
+                        v-on:update:checked="handleFilterAction(t, 'type')"/>
+              <span class="title is-5 is-size-16">{{ t.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+    <Accordeon
+          name="Domains"
+          :opened="domainAccordionOpened"
+          @update:opened="domainAccordionOpened=!domainAccordionOpened"
+        >
+          <div
+            v-for="domain in domains"
+            :key="domain.name"
+          >
+            <div class="filter-item">
+              <Checkbox :checked = "domain.checked"
+                        v-on:update:checked="handleFilterAction(domain, 'domain')"/>
+              <span class="title is-5 is-size-16">{{ domain.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+    <Accordeon
+          v-if="customers && customers.length"
+          name="Customers"
+          :opened="customerAccordionOpened"
+          @update:opened="customerAccordionOpened=!customerAccordionOpened"
+        >
+          <div
+            v-for="customer in convertedCustomers"
+            :key="customer.name"
+          >
+            <div class="filter-item">
+              <Checkbox :checked = "customer.checked"
+                        v-on:update:checked="handleFilterAction(customer, 'customer')" /> 
+              <span class="title is-5 is-size-16">{{ customer.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import {IProject} from '../../../shared/interfaces/project';
+  import Accordeon from '../../common/Accordeon/Accordeon.vue';
+  import Checkbox from '../../common/Checkbox/Checkbox.vue';
+  import Constants from '../../../shared/constants/constants';
+  import { Extension } from '../../../shared/classes/Extension';
+  
+
+  export default Vue.extend({
+      data() {
+        return {
+            // properties for accordion sections open/closed
+            lineAccordionOpened: true,
+            programAccordionOpened: true,
+            typeAccordionOpened: true,
+            domainAccordionOpened: true,
+            customerAccordionOpened: true,
+            technologyAccordionOpened: true,
+
+            // models for accordeon section
+            lines: new Array(),
+            domains: new Array(),
+            programs: new Array(),
+            types: new Array()
```

Comment:
See above - no special arrays and booleans should be present, bu a map like -
[{opened: true, items: [], opened: true, items: [], opened: true, items: [], opened: true, items: []} ]